uptane-standard.md

```diff
@@ -289,20 +289,309 @@ The Timestamp role SHALL produce and sign metadata indicating whether there are
+
+Note that a role name SHOULD follow the filename restrictions of the underlying file storage mechanism. For example, it may be “director” or "targets/director." As discussed in Section 3.7, the role name will determine part of the actual metadata filename of the delegated targets role. If it is “director” or “targets/director,” then its delegated targets metadata file MAY use the filename “director.ext” or “targets/director.ext,” respectively. However, the role name SHALL NOT use the path separator (e.g., “/” or “\”) if it is a character used to separate directories on the underlying file storage mechanism. In other words, all targets metadata files are implicitly assumed to reside in the same directory. It is safe to use this character in key-value databases or stores that do not have a notion of directories (e.g., Amazon S3).
+
+Finally, the `PathsToRoles.terminating` attribute determines whether or not a backtracking search for a target/image should be terminated.
+
+The metadata file for a delegated targets role SHALL have exactly the same format as for the top-level targets role. For example, the metadata file for a supplier role has precisely the same format as the the top-level targets role.
+
 ### Snapshot Metadata {#snapshot_meta}
 
+The snapshot metadata lists the version numbers of all targets metadata files on the repository. It is signed using the snapshot role keys, and follows the format specified here.
+```
+; Adjust length to your needs.
+SnapshotMetadata              = NumberOfSnapshotMetadataFiles SnapshotMetadataFiles
+NumberOfSnapshotMetadataFiles = Length
+SnapshotMetadataFiles         = 1*128SnapshotMetadataFile
+SnapshotMetadataFile          = StrictFilename Version
+; https://tools.ietf.org/html/rfc6025#section-2.4.2
+```
+The `filename` attribute specifies a metadata file's relative path from the metadata root of a repository, and SHALL NOT contain a path separator.
+
+An ECU SHOULD verify that each filename has been defined exactly once in the snapshot metadata file.
+
 ### Timestamp Metadata {#timestamp_meta}
 
+The timestamp metadata specifies metadata (e.g., filename and version number) about the snapshot metadata file. It is signed using the timestamp role keys, and follows the format below.
+```
+; https://tools.ietf.org/html/rfc6025#section-2.4.2
+TimestampMetadata = Filename Version Length NumberOfHashes Hashes
+NumberOfHashes = Length
+```
+
+### The map file
+
+The map file specifies which images should be downloaded from which repositories. In most deployment scenarios for full verification ECUs, this will mean downloading images from both the image and director repositories. It is not signed, and follows the format specified here.
+```
+; https://github.com/theupdateframework/taps/blob/master/tap4.md
+MapFile = NumberOfRepositories  Repositories NumberOfMappings Mappings
```

Comment:
Can we check whether this is up to date with TAP 4?

Reply:
It looks like the threshold field is missing from the mapping, I'll add that.